script.sinaweibo/resources/lib/weibo.py

```diff
@@ -0,0 +1,156 @@
+        self.teamObjs = {}
+
+    def onInit(self):
+        xbmc.log(msg="weibo center  cycle started", level=xbmc.LOGDEBUG)
+        self.getControl(32540).setImage(os.path.join(addon_path, "resources", "images", "weibo_sm.png"))
+        self.refresh()
+        i = 0
+        while self.isRunning:
+            if (float(i * 200) / (twitter_update_time * 60 * 1000)).is_integer() and ((i * 200) / (3 * 60 * 1000)) != 0:
+                self.getTweets()
+            xbmc.sleep(200)
+            i += 1
+        xbmc.log(msg="[Match Center] Twitter cycle stopped", level=xbmc.LOGDEBUG)
+
+    def getTweets(self):
+        self.getControl(32500).setLabel("#" + self.hash)
+        self.getControl(32503).setImage(os.path.join(addon_path, "resources", "images", "weibo_sm.png"))
+        tweetitems = []
+        tweets = sina_tweet.get_uid_tweets(self.hash)
+        if tweets:
+            for _tweet in tweets:
+                #td = ssutils.get_timedelta_string(datetime.datetime.utcnow() - _tweet["date"])
+                td = _tweet["date"]
+                item = xbmcgui.ListItem(_tweet["text"].replace("\n", ""))
+                item.setProperty("profilepic", _tweet["profilepic"])
+                item.setProperty("author", "[B]" + "@" + _tweet["author"] + "[/B]")
+                item.setProperty("timedelta", td)
+                tweetitems.append(item)
+
+        self.getControl(32501).reset()
+        self.getControl(32501).addItems(tweetitems)
+        if tweetitems:
+            self.setFocusId(32501)
+        return
+
+    def getSubscriptions(self):
+        self.getControl(32500).setLabel("#" + "Subscriptions")
+        self.getControl(32503).setImage(os.path.join(addon_path, "resources", "images", "weibo_sm.png"))
+        tweetitems = []
+        #self.hash = ['1784827665', '2323370715']
+        for username in self.hash:
+            tweets = sina_tweet.get_uid_tweets(username)
+            if tweets:
+                for _tweet in tweets:
+                    #td = ssutils.get_timedelta_string(datetime.datetime.utcnow() - _tweet["date"])
+                    td = _tweet["date"]
+                    item = xbmcgui.ListItem(_tweet["text"].replace("\n", ""))
+                    item.setProperty("profilepic", _tweet["profilepic"])
+                    item.setProperty("author", "[B]" + "@" + _tweet["author"] + "[/B]")
+                    item.setProperty("timedelta", td)
+                    tweetitems.append(item)
+
+        self.getControl(32501).reset()
+        self.getControl(32501).addItems(tweetitems)
+        if tweetitems:
+            self.setFocusId(32501)
+        return
+
+    def reset(self):
+        xbmcgui.Dialog().ok(translate(32000), translate(32045))
+        if os.path.exists(tweet_file):
+            os.remove(tweet_file)
+            xbmcgui.Dialog().ok(translate(32000), translate(32045))
+        return
+
+    def refresh(self):
+        xbmc.executebuiltin("SetProperty(loading-script-sinaweibo-weibo,1,home)")
```

Comment:
I think you can get the window ID and just use setProperty without needing the builtin